### DISCUSSION TITLE
Linear decomposition of displacement field in computational homogenization

### DISCUSSION THREAD

**Discussion by tairoon1** [idaholab/moose#17369] [Q&A Modules: Solid mechanics] [answered]
Hi everyone,

In first-order computational homogenization, the displacement is typically decomposed into a mean field u_bar and a fluctuation field w(X), i.e.
u(X) = u_bar(X) + w(X).

u_bar is actually fully defined everywhere on the domain with u_bar = (F_bar-I) X, where F_bar is given and I is the identity matrix, and w(X) is periodic on the boundaries. So, one is interested in finding the fluctuation field w(X).

Is it possible to do that inside TensorMechanics module and if yes, how would one write the Input file to do that?

Thanks.

**Comment by ngrilli** [comment, marked as answer]
Dear @tairoon1 

I don't think such homogenisation method is already implemented in moose,
but it can definitely be implemented by solving for two independent variables u and w.

Something similar has been done in this repository:
https://github.com/lanl/tardigrade-micromorphic-element

Best Regards,
Nicolò Grilli

**Reply by tairoon1** [reply]
Dear Nicolò,

Thanks for your quick reply. So does that mean, it is not yet possible to use MOOSE for such application? Are there any plans to support such a feature in the future?

Best regards
